src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java

```diff
@@ -29,49 +28,86 @@ public QuestionnaireLinkedService(
   }
 
   public void processQuestionnaireLinked(ResponseManagementEvent questionnaireLinkedEvent) {
-    UacDTO uac = questionnaireLinkedEvent.getPayload().getUac();
-    String questionnaireId = uac.getQuestionnaireId();
-    UacQidLink uacQidLink = uacService.findByQid(questionnaireId);
+    UacDTO uacDTO = questionnaireLinkedEvent.getPayload().getUac();
+    UacQidLink uacQidLink = uacService.findByQid(uacDTO.getQuestionnaireId());
 
-    checkQidNotLinkedToAnotherCase(uac, uacQidLink);
+    // if the uacqidLink is already linked to a dfiferent case this throws an Exception
+    checkQidNotLinkedToAnotherCase(uacDTO, uacQidLink);
 
-    Case caze;
+    logEvent(questionnaireLinkedEvent, uacQidLink);
 
-    if (isIndividualQuestionnaireType(questionnaireId)) {
-      Case householdCase = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
-      caze = caseService.prepareIndividualResponseCaseFromParentCase(householdCase);
+    Case caze = getCase(uacDTO);
+    uacQidLink.setCaze(caze);
+
+    if (uacQidLink.isBlankQuestionnaireReceived()) {
+      processBlankQuestionnaireLinked(uacQidLink);
+      return;
+    }
 
-      caseService.emitCaseCreatedEvent(caze);
-    } else {
-      caze = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
+    if (caze.isCcsCase()) {
+      processCCSCaseLinking(caze, uacQidLink);
+      return;
     }
 
-    // If UAC/QID has been receipted before case, update case
+    processStandardLinking(uacQidLink, caze);
```

Comment:
this is good. Let's see if we can get ourselves to a place where we get rid of all the CCS c**p and this method can just be called `processLinking`